src/appleseed/foundation/utility/iesparser.h

```diff
@@ -0,0 +1,459 @@
+
+    // Parse all photometric data after the TILT section
+    void parse_photometric_data(std::istream& input_stream);
+
+    void parse_angles(std::istream& input_stream);
+
+    void parse_candela_values(std::istream& input_stream);
+
+    // Check if the line defines a valid keyword-value pair
+    // and the keyword passes the used standard requirements.
+    static bool is_keyword_line(const std::string& line);
+
+    // Check if the line is a valid TILT=<...> line.
+    static bool is_tilt_line(const std::string& line);
+
+    // Parse the line containing keyword-value pair
+    // and store this pair in the dictionary.
+    void parse_keyword_line(const std::string& line);
+
+    // Parse TILT=<...> line.
+    void parse_tilt_line(const std::string& line);
+
+    // Process BLOCK and ENDBLOCK keywords.
+    void process_block_keyword(const std::string& keyword);
+
+    // Check if the specified standard allows this keyword.
+    void accept_keyword(const std::string& keyword);
+
+    // Check if all required keywords were found.
+    void check_required_keywords() const;
+
+    void check_iesna02_required_keywords() const;
+
+    void check_iesna91_required_keywords() const;
+
+    // Check if line is not empty and EOF is not reached.
+    void check_empty(std::istream& input_stream) const;
+
+    // Read lines, split each line and write the values to the vector,
+    // until the total number of values does not exceed the specified size.
+    // If number of values in the last parsed line it too large, raise a ParsingException.
+    // Can also raise a boost::bad_cast exception, when a token cannot be converted
+    // to the value of specified type.
+    template <typename ValueType>
+    std::vector<ValueType> parse_to_vector(
+        std::istream& input_stream, size_t count)
+    {
+        std::vector<ValueType> output;
+        output.reserve(count);
+        while (output.size() < count)
+        {
+            check_empty(input_stream);
+
+            std::vector<std::string> tokens;
+            boost::split(tokens, line, isspace, boost::token_compress_on);
+            if (output.size() + tokens.size() > count)
+            {
+                std::string expected_number_of_values =
+                    boost::lexical_cast<std::string>(count - output.size());
+                throw ParsingException(
+                    ("Too many values in the line, expected " + 
+                    expected_number_of_values).c_str(), line_counter);
+            }
+
+            for (each< std::vector<std::string> > token = tokens; token; ++token)
+            {
+                output.push_back(
+                    boost::lexical_cast<ValueType>(*token));
+            }
+
+            read_trimmed_line(input_stream);
+        }
+
+        return output;
+    }
+
+    //
+    // All information retrieved from file:
+    //
+
+    // File format.
+    Format format;
+
+    // TILT specification.
+    TiltSpecification tilt_specification;
```

Comment:
One specificity of our formatting convention is that we vertically align class data members:

    Type1      m_member1;
    LongType2  m_member2;
    ...